components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -2782,65 +2775,45 @@ void RewardsServiceImpl::OnRestorePublishersInternal(
   callback(result);
 }
 
-std::unique_ptr<ledger::PublisherInfoList>
-SaveNormalizedPublisherListOnFileTaskRunner(PublisherInfoDatabase* backend,
-    const ledger::PublisherInfoList& list) {
+bool SaveNormalizedPublisherListOnFileTaskRunner(
+    PublisherInfoDatabase* backend,
+    ledger::PublisherInfoList list) {
   if (!backend) {
-    return nullptr;
-  }
-
-  bool success = backend->InsertOrUpdateActivityInfos(list);
-
-  if (!success) {
-    return nullptr;
-  }
-
-  std::unique_ptr<ledger::PublisherInfoList> new_list(
-      new ledger::PublisherInfoList);
-
-  for (auto& publisher : list) {
-    new_list->push_back(publisher);
+    return false;
   }
 
-  return new_list;
+  return backend->InsertOrUpdateActivityInfos(list);
 }
 
 void RewardsServiceImpl::SaveNormalizedPublisherList(
-      const ledger::PublisherInfoListStruct& list) {
-  if (list.list.size() == 0) {
-    std::unique_ptr<ledger::PublisherInfoList> empty_list(
-        new ledger::PublisherInfoList);
-    OnPublisherListNormalizedSaved(std::move(empty_list));
-    return;
+    ledger::PublisherInfoList list) {
+  ContentSiteList site_list;
+  for (const auto& publisher : list) {
+    site_list.push_back(PublisherInfoToContentSite(*publisher));
   }
+  // TODO(bridiver) - this doesn't belong here
+  std::sort(site_list.begin(), site_list.end());
 
   base::PostTaskAndReplyWithResult(
-    file_task_runner_.get(),
-    FROM_HERE,
-    base::Bind(&SaveNormalizedPublisherListOnFileTaskRunner,
-               publisher_info_backend_.get(),
-               list.list),
-    base::Bind(&RewardsServiceImpl::OnPublisherListNormalizedSaved,
-               AsWeakPtr()));
+      file_task_runner_.get(),
+      FROM_HERE,
+      base::BindOnce(&SaveNormalizedPublisherListOnFileTaskRunner,
+                     publisher_info_backend_.get(),
+                     std::move(list)),
+      base::BindOnce(&RewardsServiceImpl::OnPublisherListNormalizedSaved,
+                     AsWeakPtr(),
+                     std::move(site_list)));
 }
 
 void RewardsServiceImpl::OnPublisherListNormalizedSaved(
-    std::unique_ptr<ledger::PublisherInfoList> list) {
-  if (!list) {
+    ContentSiteList site_list,
+    bool success) {
+  if (!success) {
     LOG(ERROR) << "Problem saving normalized publishers "
                   "in SaveNormalizedPublisherList";
     return;
   }
 
-  ContentSiteList site_list;
-  for (auto& publisher : *list) {
-    if (publisher.percent >= 1) {
-      site_list.push_back(PublisherInfoToContentSite(publisher));
```

Comment:
is these two lines supposed to be removed by this PR, or we moved this logic somewhere else?

Reply:
https://github.com/brave/brave-core/pull/2432/files/a78418feaeff19e8a2644121f1297db34199955a#diff-a5d16be9c9b52a6df7b866b1d75d3f5fR2790

Reply:
As we discussed offline, this logic (the `if (publisher.percent >= 1)` part) was unintentionally removed when rebasing, let's put it back for now and move this UI logic to a suitable place later.

Reply:
this is fixed